extract-metadata.py

```diff
@@ -0,0 +1,71 @@
+from dandi.dandiapi import DandiAPIClient
+from dandi.dandiapi import RemoteDandiset
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+client = DandiAPIClient("https://api.lincbrain.org/api")
+client.dandi_authenticate()
+
+# Create dataframe of all assets across all Dandisets
+dandisets = []
+for data in client.paginate("/dandisets/"):
+    dandisets.append(RemoteDandiset.from_data(client, data))
+print(f"Number of Dandisets: {len(dandisets)}")
+
+df = pd.DataFrame(columns=["Dandiset",
+                           "Version",
+                           "Subject", 
+                           "Modality",
+                           "Path",
+                           "Filename",
+                           "Extension",
+                           "Directory", # Top-level directory (e.g. source data, raw data, derivatives)
+                           'Size (bytes)'])
+
+modalities = {'oct': 'PS-OCT',
+              'df': 'Dark Field Microscopy',
+              'xpct': 'HiP-CT',
+              'dwi': 'DWI',
+              'fluo': 'LSM'}
+
+for dandiset in client.get_dandisets():
+    latest_dandiset = dandiset.for_version('draft')
+    for asset in latest_dandiset.get_assets():
+
+        metadata = asset.get_metadata()
+        metadata_dict = metadata.model_dump(mode='json', exclude_none=True)
+
+        subject = 'Unknown'
+        for part in asset.path.split('/'):
+            if part.startswith("sub-"):
```

Comment:
Is this line pretty brittle if our naming conventions dont always follow a `sub-` prefix?

Reply:
Thats a fair point.  I have kept it this way so that we know which assets adhere to the BIDS convention.  According to BIDS, most assets should have the `sub-` entity in the path name.  I have added an `if` clause to capture those assets that may not be associated with a single subject (e.g. `dataset_description.json`, `participants.tsv`, `readme.md`, `samples.tsv`).  See [commit](https://github.com/lincbrain/linc-dashboard/pull/4/commits/a31cfe5faaf259e69e3daf0e2eb4aca4bb53ded9).